kernel/nano_core/src/lib.rs

```diff
@@ -120,11 +120,16 @@ where
     let rsdp_address = boot_info.rsdp();
     println!("nano_core(): bootloader-provided RSDP address: {:X?}", rsdp_address);
 
-    // If the bootloader already mapped the framebuffer for us, the we can use it now
-    // before initializing the memory mgmt subsystem.
-    let framebuffer_info = boot_info.framebuffer_info();
-    if let Some(ref fb_info) = framebuffer_info && fb_info.is_mapped() {
-        early_printer::init(fb_info)?;
+    #[cfg(target_arch = "x86_64")]
+    let framebuffer_info;
+
+    #[cfg(target_arch = "x86_64")] {
+        // If the bootloader already mapped the framebuffer for us, the we can use it now
+        // before initializing the memory mgmt subsystem.
+        framebuffer_info = boot_info.framebuffer_info();
+        if let Some(ref fb_info) = framebuffer_info && fb_info.is_mapped() {
+            early_printer::init(fb_info)?;
+        }
```

Comment:
i think this is a bit cleaner and easier to read
```suggestion
    // If the bootloader already mapped the framebuffer for us, the we can use it now
    // before initializing the memory mgmt subsystem.
    #[cfg(target_arch = "x86_64")]
    let framebuffer_info = boot_info.framebuffer_info();
    #[cfg(target_arch = "x86_64")]
    if let Some(ref fb_info) = framebuffer_info && fb_info.is_mapped() {
        early_printer::init(fb_info)?;
```

Reply:
Also, let's refactor this PR a bit -- move the changes to nano_core into a separate PR, since it has nothing to do with multicore support.